base/compiler/optimize.jl

```diff
@@ -386,6 +374,14 @@ function inline_worthy(body::Array{Any,1}, src::CodeInfo, sptypes::Vector{Any},
             # but assume that forward jumps are already counted for from
             # summing the cost of the not-taken branch
             thiscost = stmt.label < line ? 40 : 0
+        elseif stmt isa GotoIfNot
+            thiscost = stmt.dest < line ? 40 : 0
+        elseif stmt isa ReturnNode
+            if stmt.val isa Expr
+                thiscost = statement_cost(stmt.val, -1, src, sptypes, slottypes, params)::Int
+            else
```

Comment:
Shouldn't this be unreachable now (malformed IR)?

Reply:
I guess so. There can still be some `Expr`s there, but I guess they all have 0 cost.